Implement new in-page alerts design

bl-nero opened the pull request:
<img width="633" alt="Screenshot 2024-07-31 at 19 40 15" src="https://github.com/user-attachments/assets/041304a8-da08-4010-b5a3-da71e7261c5f">
<img width="738" alt="Screenshot 2024-07-31 at 19 40 25" src="https://github.com/user-attachments/assets/023ae954-55cf-412c-979e-c78b9ca46ec3">

Needs to be merged along with https://github.com/gravitational/teleport.e/pull/4810.

Figma: https://www.figma.com/design/Gpjs9vjhzUKF1GDbeG9JGE/Application-Design-System?node-id=8890-5504&m=dev

Closes https://github.com/gravitational/teleport/issues/44596

bl-nero committed:
Replace old in-page alerts with the new design

bl-nero committed:
Update snapshots

gzdunek reviewed (approved):
The new alerts look so much better!

bl-nero commented on web/packages/design/src/Alert/Alert.tsx:
```diff
@@ -38,79 +43,145 @@ type Kind =
+  );
+};
+
+Alert.displayName = 'Alert';
```
I don't know. The only case I noticed when it ever made a difference is in the React dev tools in production (minimized) mode. Since I don't have a strong opinion here, and I see these used in quite a few places in the codebase, I'd rather leave it in places where it's been defined (unless you _know_ it's not needed, in which case I'll gladly remove it).

gzdunek replied:
I believe we can safely remove it https://legacy.reactjs.org/docs/react-component.html#displayname.

bl-nero committed:
Remove a commented out test

roraback commented on web/packages/design/src/Alert/Alert.tsx:
```diff
@@ -121,19 +192,107 @@ const Alert = styled.div<AlertProps>`
+    case 'outline-warn':
+      return { fill: 'border', intent: 'neutral' };
+    case 'success':
+      return { fill: 'filled', intent: 'neutral' };
```
Thanks for catching this, @gzdunek! It was a mistake in the design file. This should be a border button.

bl-nero replied:
Well, well, it _was_ a mistake. Congratulations on hawk eye, @gzdunek!

kimlisa commented on e:
wasn't sure if this was intended

bl-nero replied:
Daaamn, thanks for noticing this. Yes, indeed, the UI is meant to be built using transparency according to the design. The "hovering" state was, I believe, my own invention back in the unified resource project days. I think I'll work around it by adding a regular background below it in this particular corner case.

kimlisa reviewed (commented):
looks good! do you mind taking a look into this:

since the background is transparent, this error becomes hard to read
https://localhost:9002/?path=/story/shared-unifiedresources--failed-after-scrolling&globals=theme:Light%20Theme
<img width="1411" alt="image" src="https://github.com/user-attachments/assets/c930d7c7-cf87-4e69-adb0-7f9449b3b1d0">

https://localhost:9002/?path=/story/shared-unifiedresources--failed&globals=theme:Light%20Theme
<img width="501" alt="image" src="https://github.com/user-attachments/assets/135296df-7762-4a6c-a305-c92cb3c1ebaa">


https://localhost:9002/?path=/story/teleporte-accessrequests-notificationroutingrules--with-plugin-error&globals=theme:Light+Theme
<img width="1055" alt="image" src="https://github.com/user-attachments/assets/c666fe58-fcdc-49a2-9efe-6dbc7f18d304">


i realized that for this, i didn't add an exclusive story for error state 🤔 , it errored out b/c msw being occasionally being wonky...
https://localhost:9002/?path=/story/teleporte-integrations-enroll-okta--import-init-loading&globals=theme:Light+Theme
<img width="1315" alt="image" src="https://github.com/user-attachments/assets/bb79c7d1-c7ae-4d99-8ee2-0620d7dac559">

kimlisa reviewed (approved):
after addressing comments lgtm

bl-nero committed:
Review

bl-nero commented:
@kimlisa Thanks for noticing the transparency issue. Yes, indeed, the UI is meant to be built using transparency according to the design. The "hovering" state was, I believe, my own invention back in the unified resource project days. I think I'll work around it by adding a regular background below it in this particular corner case.

bl-nero committed:
Update snapshots

bl-nero commented:
Issues mentioned by @kimlisa are fixed. The transparency one was quite involved, so I'd encourage one more look at it. I also made use of the `details` prop to properly display the error details in unified resources view.

![Screenshot 2024-08-02 at 19 11 49](https://github.com/user-attachments/assets/fcb327ed-1454-4eea-ac23-112188671ee5)
![Screenshot 2024-08-02 at 19 02 37](https://github.com/user-attachments/assets/a4be34dc-9eb6-4980-bfc7-d99ca48861ca)
![Screenshot 2024-08-02 at 18 52 34](https://github.com/user-attachments/assets/8861289b-cdac-4d85-874c-5af5f1055e60)

kimlisa reviewed (approved):
thanks for changes! one last find:

https://localhost:9002/?path=/story/teleporte-accessrequests-notificationroutingrules--with-list-rule-errors
<img width="784" alt="image" src="https://github.com/user-attachments/assets/b0302af3-2ec5-4b94-8b19-e1ae7a033921">

bl-nero committed:
Merge branch 'master' into bl-nero/alerts

bl-nero committed:
Fix the alert in the DB auto-enrollment

bl-nero commented:
I fixed one more (and updated the story), @kimlisa FYI:

![Screenshot 2024-08-05 at 13 52 59](https://github.com/user-attachments/assets/3570cea6-21cb-4605-a8f7-ccbde3b51986)
